apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -216,49 +226,101 @@ Keep in mind you are not just generating an example; you are generating an actua
       user: openAiUserFromAuthn(authnUserId),
     });
 
-    extractFromCompletion(completion, job);
-    const html = job?.data?.html;
-
-    job.data['initialGenerationErrors'] = [];
-    job.data['prompt'] = userPrompt;
-    job.data['generation'] = completion.choices[0].message.content;
-    job.data['context'] = context;
-    job.data['completion'] = completion;
+    const results = extractFromCompletion(completion, job);
+    const html = results?.html;
 
+    let errors: string[] = [];
     if (html && typeof html === 'string') {
-      const errors = validateHTML(html, false, !!job?.data?.python);
-      job.data['initialGenerationErrors'] = errors;
-      job.data['finalGenerationErrors'] = errors;
-      if (errors.length > 0) {
-        await regenInternal(
-          job,
-          client,
-          authnUserId,
-          userPrompt,
-          `Please fix the following issues: \n${errors.join('\n')}`,
-          html,
-          typeof job?.data?.python === 'string' ? job?.data?.python : undefined,
-          0,
-          false,
-        );
-      }
+      errors = validateHTML(html, false, !!results?.python);
     } else {
+      errors = ['Please generate a question.html file.'];
+    }
+
+    let questionQid = '';
+
+    if (saveLocals) {
+      const files = {};
+
+      if (results?.html) {
+        files['question.html'] = results?.html;
+      }
+
+      if (results?.python) {
+        files['server.py'] = results?.python;
+      }
+
+      const draftId = await queryRow(sql.update_draft_number, { course_id: courseId }, z.number());
```

Comment:
Is this an ID or a number? (It's a number - "id" pretty much always refers to a Postgres primary key.)

Reply:
changed in ead40f9.